utils/state_graph.py

```diff
@@ -25,10 +30,34 @@ def generate(state: State):
         docs_content = "\n\n".join(
             doc.page_content for doc in state["context"]
         )
+
+        # Debug log context preparation
+        logger.info(f"🧠 DEBUG: Preparing context from {len(state['context'])} documents")
+        logger.info(f"📝 DEBUG: Total context length: {len(docs_content)} characters")
+
+        # Show a preview of the context being sent to LLM
+        context_preview = docs_content[:500].replace('\n', ' ')
+        logger.info(f"📜 DEBUG: Context preview: {context_preview}...")
+
         messages = prompt.invoke(
             {"question": state["question"], "context": docs_content}
         )
+
+        # Debug log prompt preparation
+        if hasattr(messages, 'content'):
+            prompt_preview = str(messages.content)[:300].replace('\n', ' ')
+            logger.info(f"📝 DEBUG: Prompt preview: {prompt_preview}...")
```

Comment:
should we add these as debug lines instead of info and add a command line option to look at debug logs?